tests/test_yaml_search_space/test_search_space.py

```diff
@@ -8,105 +8,64 @@
-        assert pipeline_space["param_int1"].default_confidence_score == 0.5
-        assert isinstance(pipeline_space["param_int2"], IntegerParameter)
-        assert pipeline_space["param_int2"].lower == 100
-        assert pipeline_space["param_int2"].upper == 30000
-        assert pipeline_space["param_int2"].log is True
-        assert pipeline_space["param_int2"].is_fidelity is False
-        assert pipeline_space["param_int2"].default is None
-        assert pipeline_space["param_int2"].default_confidence_score == 0.5
-        assert isinstance(pipeline_space["param_float2"], FloatParameter)
-        assert pipeline_space["param_float2"].lower == 3.3e-5
-        assert pipeline_space["param_float2"].upper == 0.15
-        assert pipeline_space["param_float2"].log is False
-        assert pipeline_space["param_float2"].is_fidelity is False
-        assert pipeline_space["param_float2"].default is None
-        assert pipeline_space["param_float2"].default_confidence_score == 0.5
-        assert isinstance(pipeline_space["param_cat"], CategoricalParameter)
-        assert pipeline_space["param_cat"].choices == [2, "sgd", 10e-3]
-        assert pipeline_space["param_cat"].is_fidelity is False
-        assert pipeline_space["param_cat"].default is None
-        assert pipeline_space["param_cat"].default_confidence_score == 2
-        assert isinstance(pipeline_space["param_const1"], ConstantParameter)
-        assert pipeline_space["param_const1"].value == 0.5
-        assert pipeline_space["param_const1"].is_fidelity is False
-        assert isinstance(pipeline_space["param_const2"], ConstantParameter)
-        assert pipeline_space["param_const2"].value == 1e3
-        assert pipeline_space["param_const2"].is_fidelity is True
-
-    test_correct_yaml_file("tests/test_yaml_search_space/correct_config.yaml")
-    test_correct_yaml_file(
-        "tests/test_yaml_search_space/correct_config_including_types" ".yaml"
-    )
+        float1 = FloatParameter(0.00001, 0.1, True, False)
+        assert float1.__eq__(pipeline_space["param_float1"]) is True
+        int1 = IntegerParameter(-3, 30, False, True)
+        assert int1.__eq__(pipeline_space["param_int1"]) is True
+        int2 = IntegerParameter(100, 30000, True, False)
+        assert int2.__eq__(pipeline_space["param_int2"]) is True
+        float2 = FloatParameter(3.3e-5, 0.15, False, False)
+        assert float2.__eq__(pipeline_space["param_float2"]) is True
+        cat1 = CategoricalParameter([2, "sgd", 10e-3], False)
+        assert cat1.__eq__(pipeline_space["param_cat"]) is True
+        const1 = ConstantParameter(0.5, False)
+        assert const1.__eq__(pipeline_space["param_const1"]) is True
+        const2 = ConstantParameter(1e3, True)
+        assert const2.__eq__(pipeline_space["param_const2"]) is True
+
+    test_correct_yaml_file(BASE_PATH + "correct_config.yaml")
+    test_correct_yaml_file(BASE_PATH + "correct_config_including_types.yaml")
```

Comment:
Three things on code quality improvements:
1. `assert something() is True` is equivalent to `assert something()`
2. `a.__eq__(b)` is equivalent to `a == b`
3. You can use `pytest.mark.parametrize(...)` to essentially run this test twice, each with a different yaml file.

I would personally write this in the following way:
```python
@pytest.mark.neps_api
@pytest.mark.parametrize(
	"path",
	[
		BASE_PATH + "correct_config.yaml",
		BASE_PATH + "correct_config_including_types.yaml"
	]
)
def test_correct_yaml_files(path: str):
	"""Test the function with a correctly formatted YAML file."""
	pipeline_space = pipeline_space_from_yaml(path)
	assert isinstance(pipeline_space, dict)
	
	float1 = FloatParameter(0.00001, 0.1, True, False)
	int1 = IntegerParameter(-3, 30, False, True)
	int2 = IntegerParameter(100, 30000, True, False)
	cat1 = CategoricalParameter([2, "sgd", 10e-3], False)
	float2 = FloatParameter(3.3e-5, 0.15, False, False)
	const1 = ConstantParameter(0.5, False)
	const2 = ConstantParameter(1e3, True)

	assert pipeline_space["param_float1"] == float1
	assert pipeline_space["param_int1"] == int1
	assert pipeline_space["param_int2"] == int2	
	assert pipeline_space["param_float2"]) == float2
	assert pipeline_space["param_cat"] == cat1
	assert pipeline_space["param_const1"] == const1
	assert pipeline_space["param_const2"] == const2
```